src/charm.py

```diff
@@ -166,14 +173,14 @@ def _configure_nrf(self, event: EventBase) -> None:
         self._publish_nrf_info_for_all_requirers()
         self.unit.status = ActiveStatus()
 
-    def _on_certificates_relation_created(self, event: EventBase) -> None:
+    def _on_certificates_relation_created(self, event: RelationCreatedEvent) -> None:
```

Comment:
@ghislainbourgeois encouraged us to use the more generic `EventBase` though I must confess I don't remember the reason. I'm not sure now's the right time to revert to the more specific events. At least let's all agree on the approach.

Reply:
Ok, I removed it from here. 

I changed this while I was poking around trying to figure out if there was any other information I could use to determine if the charm was being removed. I figured this was a pretty non-controversial change so I included it, but if it was intentional then you're right, not the place to discuss it.